astropy/io/fits/tests/test_core.py

```diff
@@ -273,13 +273,9 @@ def make_invalid_hdu():
         # silentfix+exception should only mention the unfixable error in the
         # exception
         hdu = make_invalid_hdu()
-        try:
+        with pytest.raises(fits.VerifyError , match='Illegal keyword name') as excinfo:
             hdu.verify('silentfix+exception')
-        except fits.VerifyError as exc:
-            assert 'Illegal keyword name' in str(exc)
-            assert 'not upper case' not in str(exc)
-        else:
-            self.fail('An exception should have been raised.')
+        assert 'not upper case' not in str(excinfo.value)
```

Comment:
Even though it has been the same code before as the next diff both are different now. Is there a specific reason?

Reply:
First one is checking for the pattern _`not in`_ (`silentfix`), the second for `'not upper case' in str(exc)` (non-silent `fix`). See next comment.

Reply:
Ah 🤦‍♂